test/commands/source/retrieve.test.ts

```diff
@@ -266,4 +274,106 @@ describe('force:source:retrieve', () => {
     expect(displayStub.calledOnce).to.equal(false);
     expect(getJsonStub.calledOnce).to.equal(true);
   });
+
+  it('should warn users when retrieving CustomField with --metadata', async () => {
+    const metadata = 'CustomField';
+    buildComponentSetStub.restore();
+    buildComponentSetStub = stubMethod(sandbox, ComponentSetBuilder, 'build').resolves({
+      retrieve: retrieveStub,
+      getPackageXml: () => packageXml,
+      toArray: () => {
+        return [exampleSourceComponent];
+      },
+      has: (component: ComponentLike) => {
+        expect(component).to.be.a.a('object').and.to.have.property('type');
```

Comment:
NIT: are there supposed to be two `a`'s? `to.be.a.a(`?

it's in the other tests too

Reply:
no there are not, yet the test passed o_0